### ISSUE TITLE
Invalid source control status when `git add --intent-to-add` was used

### ISSUE THREAD

**Issue by Zeroji** [microsoft/vscode#160899]
Type: <b>Bug</b>

Minimal example reproducing the issue (all extensions disabled): this repo has two files, `README` which has been renamed to `README.md` (using `mv`), and `otherfile` which has been modified in the editor. The source control pane displays correct information with regards to git status:
```
$ git status --porcelain=v1 -uall
 D README
 M otherfile
?? README.md
```
![image](https://user-images.githubusercontent.com/15339858/190162998-026546cc-027c-49e2-82a3-a6ba10ce9a0c.png)

When `git add --intent-to-add` is used to have `README.md` tracked but not yet staged, the source control pane displays incorrect info:
```
$ git add --intent-to-add README.md
$ git status --porcelain=v1 -uall
 R README -> README.md
 M otherfile
```
![image](https://user-images.githubusercontent.com/15339858/190163578-164bade8-d5c6-47a7-95fc-66c09e64ae34.png)

Clicking the file labeled "DME" opens a new editor for an empty file named "DME" that's in the index, and clicking "Open File" says the file cannot be found.

If I remove the modifications in `otherfile`, the display becomes empty:
```
$ git checkout -- otherfile
$ git status --porcelain=v1 -uall
 R README -> README.md
```
![image](https://user-images.githubusercontent.com/15339858/190164484-ce125fe4-7f83-443a-b039-16a8f6e321c8.png)

I am using `git version 2.37.1.windows.1` in Git Bash, and `git version 2.25.1` in WSL, both have the same correct output. The status pane is broken in a larger repository as well, making me unable to view diffs of renamed files before staging them.

Autogenerated system information below:

VS Code version: Code 1.71.1 (e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb, 2022-09-08T19:56:36.424Z)
OS version: Windows_NT x64 10.0.19044
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) W-2102 CPU @ 2.90GHz (4 x 2904)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.69GB (5.55GB free)|
|Process Argv|--crash-reporter-id 80cbf936-5239-476f-8010-bcdbc2a60b84|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30557514
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
pylanb8912:30545647
vsclangdf:30486550
c4g48928:30535728
hb751961:30553087
dsvsc012cf:30540253
azure-dev_surveyone:30548225
2144e591:30553903
fc301958:30567733

```

</details>

**Comment by lszomoru** [comment]
This seems to be an issue with the GitStatusParser. Will look at this early February.

**Comment by hediet** [comment]
Seems to work now:

![Code_-_Insiders_ml87aT0l7D](https://github.com/microsoft/vscode/assets/2931520/114c3333-f3c9-4076-81ba-4f808e5f9974)

**Comment by Zeroji** [comment]
Can confirm it's fixed in version 1.79, tested with the original example. Thanks to all involved :)
